internal/checkout/source_provider.go

```diff
@@ -500,7 +500,12 @@ func (cfg *Config) Run(ctx context.Context) (retErr error) {
 	var sshKeyPath string
 	var sshKnownHostsPath string
 	var sshCommand string
+
+	// when checking out using SSH the local git config should not be updated to use the credential helper as the user provided SSH key should be used.
+	// However we still need the credential helper in the global git config for the subsequent calls to do a local merge and to fetch submodules.
+	var globalGitConfigUpdate bool
 	if useSSH {
+		globalGitConfigUpdate = true
```

Comment:
Shouldn't it be the other way around?!

```suggestion
	// when checking out using SSH the local git config should not be updated to use the credential helper as the user provided SSH key should be used.
	// However we still need the credential helper in the global git config for the subsequent calls to do a local merge and to fetch submodules.
	globalGitConfigUpdate := true
	if useSSH {
		globalGitConfigUpdate = false
```

Reply:
Closing this PR since we need to address a more global problem with invocation of the credential helper when a user is explicitly passing credentials (includes SSH key) to the action. Reported in https://cloudbees.atlassian.net/browse/CBP-24068